rj_config/include/config_client/config_client.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <rclcpp/rclcpp.hpp>
+#include <rj_msgs/msg/field_dimensions.hpp>
+#include <rj_msgs/msg/game_settings.hpp>
+#include <rj_msgs/srv/set_field_dimensions.hpp>
+#include <rj_msgs/srv/set_game_settings.hpp>
+
+namespace config_client {
+using GameSettingsMsg = rj_msgs::msg::GameSettings;
+using SetGameSettingsSrv = rj_msgs::srv::SetGameSettings;
+using SetGameSettingsReq = SetGameSettingsSrv::Request;
+
+using FieldDimensionsMsg = rj_msgs::msg::FieldDimensions;
+using SetFieldDimensionsSrv = rj_msgs::srv::SetFieldDimensions;
+using SetFieldDimensionsReq = SetFieldDimensionsSrv::Request;
+
+/**
+ * \brief Helper util that sets up the proper subscribers for
+ * receiving config updates
+ */
+class ConfigClient {
+public:
+    ConfigClient(rclcpp::Node* node);
+
+    [[nodiscard]] const GameSettingsMsg& gameSettings() const {
+        return game_settings_.value();
+    }
+
+    [[nodiscard]] const FieldDimensionsMsg& fieldDimensions() const {
+        return field_dimensions_.value();
+    }
```

Comment:
I would like a way of making a callback for when a parameter changes. This should be specifiable on a per-member basis of the `GameSettings` object.

From a design perspective, I'm still honestly not sure if this approach is the right one, rather than building off of the parameter server.

Reply:
Using the parameter server doesn't give you a callback. This does. I'll add in a way to specify a callback.

Reply:
ACK